handler/enqueue_annotation.go

```diff
@@ -112,16 +171,30 @@ func parseNamespacedName(namespacedNameString string) types.NamespacedName {
 	return types.NamespacedName{}
 }
 
-// SetOwnerAnnotation sets annotations for dependent resources that needs to be watched by namespaced Owners.
-func SetOwnerAnnotation(u *unstructured.Unstructured, owner *unstructured.Unstructured) {
-	a := u.GetAnnotations()
-	if a == nil {
-		a = map[string]string{}
+// SetWatchOwnerAnnotation is a helper method to add watch annotation-based with the owner NamespacedName and
+// schema.GroupKind to the object provided. This allows you to declare the watch annotations of an owner to an object.
+// If a annotation to the same object already exists, it'll be overwritten with the newly provided version.
+func SetOwnerAnnotation(owner, object metav1.Object, ownerGK schema.GroupKind) error {
+	if len(owner.GetName()) < 1 {
+		return fmt.Errorf("%T does not have a name, cannot call SetOwnerAnnotation", owner)
+	}
+
+	if len(ownerGK.Kind) < 1 {
+		return fmt.Errorf("Owner Kind not found, cannot call SetOwnerAnnotation")
+	}
+
+	if len(ownerGK.Group) < 1 {
+		return fmt.Errorf("Owner Group not found, cannot call SetOwnerAnnotation")
+	}
```

Comment:
Pretty sure it is valid for `Group` to be empty. See: https://github.com/kubernetes/apimachinery/blob/cc2fa4f57325fd0e917d43cfed5c7a9c287b5d17/pkg/runtime/schema/group_version.go#L133-L138